### DISCUSSION TITLE
Copy and paste file with group_empty enabled

### DISCUSSION THREAD

**Discussion by drdo** [nvim-tree/nvim-tree.lua#2075] [Q&A]
Let's say I have `group_empty = true` in my config and I have the the following collapsed directories `/foo/bar/lol`
with `foo`, `bar` and `lol` empty other than the directories specified here.

If I copy some file using `api.fs.copy.node`, how do I paste it for example inside the directory `bar`?

If I place the cursor on the line with `/foo/bar/lol` it always pastes into `lol`.

**Comment by alex-courtis** [comment]
We can do this for create, by removing `/lol`. We don't have a means  to do this for paste.

Workaround: on `/foo/bar/lol`: `a`: replace `lol/` with `empty`. `lol` will be un-collapsed.

**Comment by gegoune** [comment]
There is no good way of doing it while directories are collapsed, it's impossible for plugin to guess what user means. 

Potential solution: add an api action to toggle(& turn on/off) collapsing, user could uncollapse directories temporarily either by 
- key mapping
- cursor hold auto command
then paste into desired directory and reactivate collapsing.

**Reply by drdo** [reply]
I think there is a good way this could be implemented.
Simply paste in the directory where the cursor is.

So for example if the cursor is on the `a`  in `/foo/bar/lol` you paste it into `bar`.

**Reply by gegoune** [reply]
Ah, good point actually.

**Reply by alex-courtis** [reply]
> I think there is a good way this could be implemented. Simply paste in the directory where the cursor is.

That could work; the user can still move the cursor within a line even if they use `hijack_cursor`.

**Reply by alex-courtis** [reply]
@gegoune is the paste-at-cursor feature worthy?

I'm leaning towards the toggle...

**Reply by gegoune** [reply]
I am not sure. Another option would be to display editable input in which users could delete directories all the way to one they want to paste into. 

I am not sure what's the best in terms of ux.

**Reply by alex-courtis** [reply]
> display editable input in which users could delete directories all the way to one they want to paste into.

That's consistent with add / full rename.

**Comment by drdo** [comment]
Additionally, if you try to delete it will try to delete `foo` and does not let you pick which one you want.
